test_util/Makefile

```diff
@@ -1,70 +1,86 @@
 export TF_VAR_custom_dcos_download_path_win ?= https://downloads.dcos.io/dcos/testing/master/windows/dcos_generate_config_win.sh
 export TF_VAR_variant ?= open
-export OUTPUT_FILE ?= /dev/stdout
 
-.DEFAULT_GOAL := test
+SSH_KEY ?= ./tf-dcos-rsa.pem
 
-.PHONY: maws
-maws:
-	$(eval $(shell maws login "eng-devprod"))
+ifndef DCOS_LICENSE_CONTENTS 
+$(error DCOS_LICENSE_CONTENTS is not set)
+endif
 
-.PHONY: get-terraform
-get-terraform:
-	rm -rf ./terraform_0.11.14_linux_amd64.zip
-	wget https://releases.hashicorp.com/terraform/0.11.14/terraform_0.11.14_linux_amd64.zip
-	unzip ./terraform*.zip
-	chmod +x terraform
+################################### TARGETS ################################### 
+
+.DEFAULT_GOAL := test
 
-# TODO (dcos-team): This should not be required Open DC/OS
-.PHONY: dcos-set-license
-dcos-set-license:
-	@echo ${DCOS_LICENSE_CONTENTS} > license.txt
+license.txt:
+	@echo ${DCOS_LICENSE_CONTENTS} > $@
 
 .PHONY: terraform-init
-terraform-init:  get-terraform dcos-set-license
-	./terraform init --upgrade
+terraform-init: license.txt $(TERRAFORM)
+	$(TERRAFORM) init --upgrade
 
-.PHONY: ssh-agent
-ssh-agent:
-	if [ -z "$$SSH_AUTH_SOCK" ] ; then \
-	  eval $$(ssh-agent -s); \
-	fi
+$(SSH_KEY):
+	ssh-keygen -t rsa -b 2048 -f $@ -q -N ""; \
 
-.PHONY: ssh-keygen-add
-ssh-keygen-add: ssh-agent
-	ssh-keygen -t rsa -b 2048 -f ./tf-dcos-rsa.pem -q -N ""; \
-	ssh-add ./tf-dcos-rsa.pem;
+.PHONY: cluster
+cluster: terraform-init $(SSH_KEY)
+	if [ -z "$$SSH_AUTH_SOCK" ]; then eval $$(ssh-agent -s); fi; \
+	ssh-add "$(SSH_KEY)"; \
+	$(TERRAFORM) apply -auto-approve -var ssh_public_key_file="$(SSH_KEY).pub" -var windowsagent_num=1
+
+cluster.json: cluster
+	$(TERRAFORM) output -json > $@ 
 
 .PHONY: test
-test: terraform-init ssh-keygen-add
-	./terraform apply -auto-approve
+test: cluster.json
+	ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i $(SSH_KEY) centos@$(shell jq '.masters_public_ip.value' $<) -- \
+		"source /opt/mesosphere/environment.export && cd /opt/mesosphere/active/dcos-integration-test && \
+		MASTER_PUBLIC_IP=$(shell jq '.masters_public_ip.value' $<) \
+		MASTERS_PRIVATE_IPS=$(shell jq '.masters_private_ip.value' $<) \
+		MASTER_HOSTS=$(shell jq '.masters_private_ip.value' $<) \
+		PRIVATE_AGENTS_PRIVATE_IPS=$(shell jq '.private_agent_ips.value' $<) \
+		SLAVE_HOSTS=$(shell jq '.private_agent_ips.value' $<) \
+		PUBLIC_AGENTS_PRIVATE_IPS=$(shell jq '.public_agent_ips.value' $<) \
+		PUBLIC_SLAVE_HOSTS=$(shell jq '.public_agent_ips.value' $<) \
+		pytest --full-trace --log-level=DEBUG --windows-only"
 
-.PHONY: output
-output:
-	./terraform output > ${OUTPUT_FILE}
+.PHONY: ssh 
+ssh: cluster.json
+	ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i $(SSH_KEY) centos@$(shell jq '.masters_public_ip.value' $<)
 
 .PHONY: destroy
 destroy:
-	./terraform destroy
+	$(TERRAFORM) destroy -auto-approve;
```

Comment:
Due to the error: 
`[09:41:56][Step 3/3] ${var.ssh_public_key_file == "" ? var.ssh_public_key : file(local.ssh_public_key_file)}`

don't we need to add here as well the public key like:
$(TERRAFORM) destroy -auto-approve ssh_public_key_file="$(SSH_KEY).pub";